### PR TITLE
Add 2.4.1-1-dev

### PR DESCRIPTION
**Pull request by jedcunningham**
**What this PR does / why we need it**:

**Special notes for your reviewer**:

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]

- [ ] If a new distribution or new version of Airflow is added, please make sure:
  + [ ] to follow all of the directions in the `README.md`
  + [ ] that there are Dockerfiles in all base image directories
  + [ ] that all of the new or modified Dockerfiles build successfully
- [ ] If changing an existing image, please add an applicable test in `.circleci/bin/test-airflow-image.py`

### CODE REVIEW

File: .circleci/config.yml.j2 [hunk 1]
```diff
@@ -55,7 +55,11 @@ workflows:
           extra_args: |-
             {#- If you modify this, make sure you also modify it in the 'nightly' workflow -#}
             {%- if not edge_build %}
+            {%- if airflow_version_wout_dev >= "2.4.0" %}
+            --build-arg VERSION=$(curl https://pip.astronomer.io/v2/apache-airflow/latest-{{ airflow_version_wout_dev }}.build)
+            {%- else %}
             --build-arg VERSION=$(curl https://pip.astronomer.io/simple/astronomer-certified/latest-{{ airflow_version_wout_dev }}.build)
+            {%- endif %}{# version #}
```
Comment (blag) [inline .circleci/config.yml.j2:62]:
```suggestion
            {%- endif %}{# airflow_version_wout_dev >= "2.4.0" #}
```

File: .circleci/config.yml.j2 [hunk 2]
```diff
@@ -350,7 +354,11 @@ workflows:
           extra_args: |-
             {#- If you modify this, make sure you also modify it in the 'certified-airflow' workflow -#}
             {%- if not edge_build %}
+            {%- if airflow_version_wout_dev >= "2.4.0" %}
+            --build-arg VERSION=$(curl https://pip.astronomer.io/v2/apache-airflow/latest-{{ airflow_version_wout_dev }}.build)
+            {%- else %}
             --build-arg VERSION=$(curl https://pip.astronomer.io/simple/astronomer-certified/latest-{{ airflow_version_wout_dev }}.build)
+            {%- endif %}{# version #}
```
Comment (blag) [inline .circleci/config.yml.j2:361]:
```suggestion
            {%- endif %}{# airflow_version_wout_dev >= "2.4.0" #}
```

File: 2.4.1/bullseye/include/entrypoint [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  # the ENTRYPOINT command, we have to run it here, if we haven't already
+  if [[ -z "$__TINIFIED" ]]; then
+    __TINIFIED=1 exec tini -- "$0" "$@"
+  fi
+else
+  __TINIFIED=1 exec gosu "${ASTRONOMER_USER}" tini -- "$0" "$@"
+fi
+
+if [[ -n "$EXECUTOR" && -z "$AIRFLOW__CORE__EXECUTOR" ]]; then
+  # Support for puckle style of defining configs
```
Comment (blag) [inline 2.4.1/bullseye/include/entrypoint:15]: If "puckle style" isn't a typo, I'm curious what it is.

File: .circleci/config.yml.j2 (hunk 1 above)
Comment (blag) [inline .circleci/config.yml.j2:62]:
```suggestion
            {%- endif %}{# airflow_version_wout_dev >= "2.4.0" #}
```

**Review by blag — APPROVED** [review]
LGTM, feel free to disregard my suggestions - they aren't direly important.

File: 2.4.1/bullseye/include/entrypoint (hunk 3 above)
Comment (jedcunningham) [reply 2.4.1/bullseye/include/entrypoint:15]: https://github.com/puckel/docker-airflow
and
https://hub.docker.com/r/puckel/docker-airflow

It should be "puckel" I guess.

Comment (jedcunningham) [reply 2.4.1/bullseye/include/entrypoint:15]: At one point in time it was a much more common thing in the community.

Comment (jedcunningham) [inline 2.4.1/bullseye/include/entrypoint:15]:
```suggestion
  # Support for puckel style of defining configs
```

File: 2.4.1/bullseye/include/entrypoint [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  if [[ -z "$__TINIFIED" ]]; then
+    __TINIFIED=1 exec tini -- "$0" "$@"
+  fi
+else
+  __TINIFIED=1 exec gosu "${ASTRONOMER_USER}" tini -- "$0" "$@"
+fi
+
+if [[ -n "$EXECUTOR" && -z "$AIRFLOW__CORE__EXECUTOR" ]]; then
+  # Support for puckel style of defining configs
+  export AIRFLOW__CORE__EXECUTOR "${EXECUTOR}Executor"
```
Comment (blag) [inline 2.4.1/bullseye/include/entrypoint:16]: Sorry for catching this after the merge, but should this be:

```python
export AIRFLOW__CORE__EXECUTOR="${EXECUTOR}Executor"
#                             ^
```

or is it truly trying to export the two variables `AIRFLOW__CORE__EXECUTOR` (even though that is guaranteed to be blank at this point) and the variable _named_ `${EXECUTOR}Executor`?

File: 2.4.1/bullseye/include/entrypoint (hunk 3 above)
Comment (blag) [reply 2.4.1/bullseye/include/entrypoint]: TIL, thanks!

File: 2.4.1/bullseye/include/entrypoint (hunk 4 above)
Comment (jedcunningham) [reply 2.4.1/bullseye/include/entrypoint:16]: Ha, I guess that's been broken for a long time/forever?

Comment (jedcunningham) [reply 2.4.1/bullseye/include/entrypoint:16]: Feel free to PR a fix, though this might mean we can just remove this completely too.

**Pull request merged**